src/ResourceManager/HDInsight/Commands.HDInsight/ManagementCommands/AddAzureHDInsightComponentVersionCommand.cs

```diff
@@ -20,7 +20,8 @@ namespace Microsoft.Azure.Commands.HDInsight
 {
     [Cmdlet(
         VerbsCommon.Add,
-        Constants.CommandNames.AzureHDInsightComponentVersion),
+        Constants.CommandNames.AzureHDInsightComponentVersion,
+        SupportsShouldProcess = true),
```

Comment:
That is not all.  You also need to either call ConfirmAction or ShoudlProcess correctly during cmdlet execution.  See: https://gist.github.com/markcowl/338e16fe5c8bbf195aff9f8af0db585d